### DISCUSSION TITLE
Feature: single output file for input directory

### DISCUSSION THREAD

**Discussion by glebsts** [hairyhenderson/gomplate#1817] [Ideas]
Hello,
I got my hands onto reworking some existing code. It does following:
```
for f in ./dir/**/*.yaml; do (cat "\${f}"; echo) >> merged.yaml; done
gomplate -d <..some params here> -f merged.yaml > processed.yaml
```
So basically joins files in one step, and processes resulting file in other step. I was thinking how could I get rid of first step to be able to run this as single dockerized command. In docs I saw `--input-dir` which says `all files in input directory will be processed recursively`, and also `--output-dir` says `the directory structure of the input directory will be preserved`.
I checked sources, and there is `mustTogether("outputDir", "inputDir", c.OutputDir, c.InputDir)`. 
This limitation sounds a little bit artificial.

_What do you think of allowing `--input-dir` without `--output-dir` in combination with `--out` parameter to put all output to single file? Would it be aligned with tool principles?_

Would you accept PR with such functionality?
Thank you.

**Comment by hairyhenderson** [comment]
Hi @glebsts, sorry for the (very!) late response.

In your particular example (merging YAML), there's a pretty high risk of producing invalid output if the input YAML files are simply concatenated. There's no explicit guarantee that they'll produce a multi-stream document, and so the file could end up being unparseable after gomplate's done with it (duplicate keys, etc).

I'm not sure this sort of use-case can be generalized safely such that it makes sense to support it natively.

However, you could probably accomplish this with the [`file.Read`](https://docs.gomplate.ca/functions/file/#fileread) and [`tmpl.Inline`](https://docs.gomplate.ca/functions/tmpl/#tmplinline) functions. Also have a look at https://blog.hairyhenderson.ca/post/one_template_many_outputs/ (it's not the same as what you're looking for, but might give you some ideas).
